File: 김세희/k진수에서 소수 개수 구하기.js
function isPrime(num) {
  if (num < 2) {
    return false;
  }
  if (num === 2) {
    return true;
  }

  for (let i = 2; i <= Math.floor(Math.sqrt(num)); i++) {
    if (num % i === 0) {
      // 한 번이라도 나누어 졌으니 소수가 아니므로 return false
      return false;
    }
  }
  // 나눠진 수가 없다면 해당 수는 소수이므로 return true
  return true;
}

function solution(n, k) {
  const num = n.toString(k).split("");
  let ans = 0;
  let temp = "";
  let isZero = false;
  for (let i of num) {
    if (i === "0") {
      isZero = true;
      if (temp !== "") {
        let tempNum = +temp;
        if (isPrime(tempNum)) {
          ans += 1;
        }
        temp = "";
      }
    } else {
      temp += i;
    }
  }
  if (temp !== "") {
    let tempNum = +temp;
    if (isPrime(tempNum)) {
      ans += 1;
    }
  }
  return ans;
}

File: 박정은/k진수에서 소수 개수 구하기.py
def solution(n, k):
    tmp = ''
    answer = 0

    while n > k:
        tmp += str(n % k)
        n = n//k
    tmp += str(n)

    tmp = tmp[::-1]
    nums = tmp.split('0')
    nums = list(filter(None, nums))

    for num in nums:
        num = int(num)
        if isPrime(num): answer += 1

    return answer


def isPrime(n):
    if n == 1:
        return False
    else:
        for i in range(2, int(n**0.5)+1):
            if n % i == 0:
                return False
        return True
